app/controllers/organizations/adoptable_pets_controller.rb

```diff
@@ -1,48 +1,61 @@
-class Organizations::AdoptablePetsController < Organizations::BaseController
-  include ::Pagy::Backend
-
-  skip_before_action :authenticate_user!
-  skip_verify_authorized only: %i[index]
-  before_action :set_likes, only: %i[index show],
-    if: -> { allowed_to?(:index?, Like) }
-  helper_method :get_animals
-
-  def index
-    @q = authorized_scope(Pet.unadopted, with: Organizations::AdoptablePetPolicy).ransack(params[:q])
-    @pagy, paginated_adoptable_pets = pagy(
-      @q.result.includes(:adopter_applications, :matches, images_attachments: :blob),
-      limit: 9
-    )
-    @pets = paginated_adoptable_pets
-  end
+module Organizations
+  class AdoptablePetsController < Organizations::BaseController
+    include ::Pagy::Backend
+
+    skip_before_action :authenticate_user!
+    skip_verify_authorized only: %i[index]
+    before_action :set_likes, only: %i[index show],
+      if: -> { allowed_to?(:index?, Like) }
+    helper_method :get_animals
+
+    def index
+      @q = authorized_scope(
+        case params[:species]
+        when "dog"
+          Pet.Dog.unadopted
+        when "cat"
+          Pet.Cat.unadopted
+        else
+          redirect_back_or_to root_path and return
+        end,
+        with: Organizations::AdoptablePetPolicy
+      ).ransack(params[:q])
 
-  def show
-    @adoptable_pet_info = CustomPage.first&.adoptable_pet_info
-    @pet = Pet.find(params[:id])
-    authorize! @pet, with: Organizations::AdoptablePetPolicy
-
-    if current_user&.latest_form_submission
-      @adoption_application =
-        AdopterApplication.find_by(
-          pet_id: @pet.id,
-          form_submission_id: current_user.latest_form_submission.id
-        ) ||
-        @pet.adopter_applications.build(
-          form_submission: current_user.latest_form_submission
-        )
+      @pagy, paginated_adoptable_pets = pagy(
+        @q.result.includes(:adopter_applications, :matches, images_attachments: :blob),
+        limit: 9
+      )
+
+      @pets = paginated_adoptable_pets
     end
-  end
 
-  private
+    def show
+      @adoptable_pet_info = CustomPage.first&.adoptable_pet_info
+      @pet = Pet.find(params[:id])
+      authorize! @pet, with: Organizations::AdoptablePetPolicy
 
-  def get_animals
-    Pet.species.keys.to_h do |s|
-      [s, authorized_scope(Pet.where(species: s).distinct.order(:breed), with: Organizations::AdoptablePetPolicy).pluck(:breed)]
+      if current_user&.latest_form_submission
+        @adoption_application =
+          AdopterApplication.find_by(
+            pet_id: @pet.id,
+            form_submission_id: current_user.latest_form_submission.id
+          ) ||
+          @pet.adopter_applications.build(
+            form_submission: current_user.latest_form_submission
+          )
+      end
     end
-  end
 
-  def set_likes
-    likes = current_user.person.likes
-    @likes_by_id = likes.index_by(&:pet_id)
+    private
+
+    def get_animals(species)
```

Comment:
If we have a before_action that sets `@species` we can get rid of this method parameter and also use the `@species` instance variable in the index action

Reply:
I added a scope to the Pet model which I agree makes more sense.